lib/elixir/lib/calendar.ex

```diff
@@ -265,6 +265,14 @@ defmodule Calendar do
   """
   @callback valid_time?(hour, minute, second, microsecond) :: boolean
 
+  @doc """
+  Implements inspect for dates, datetimes and naive_datetimes.
+
+  The implementation must define inspect for `Date.t()`, `DateTime.t()` and
```

Comment:
According to the `@spec`, the function `inspect/2` allows any map/struct that contains the proper `time`, `date`, `naive_datetime`, or `datetime` fields. Maybe we should reword this sentence so people don't expect only `Time`, `Date`, `DateTime`, or `NaiveDateTime` structs :)

Reply:
Since this function is intended to be called by the `Inspect` protocol, I made the assumption that matching on the structs would be appropriate.  I appreciate as a callback it become public API and therefore optimistic matching may be a better strategy if thats what you recommend.

Reply:
I agree. We should match on structs but then it means we need to update the specs to say so instead of using `Calendar.date`.